app/src/main/java/com/pvp/app/worker/MealPlanWorker.kt

```diff
@@ -26,10 +35,13 @@ class MealPlanWorker @AssistedInject constructor(
 ) {
 
     override suspend fun doWork(): Result {
+
         val user = userService.user.firstOrNull()
 
         user ?: return Result.retry()
 
+        postActivityNotification()
```

Comment:
Line 53 checks if meal plan has been already generated this week, which means, checks if user already got such notification for generated plan.
```kotlin
        if (user.lastMealPlanGeneratedWeek >= week) {
            return Result.success()
        }
```

Currently user will get notification on every application start-up.

Move notification posting below that condition check, best would be to line 63 and separate with empty line from return statement as so:
```kotlin
        userService.merge(user)

        postActivityNotification()

        return Result.success()
```